### PR TITLE
Adding Star/Unstar Methods

### PR DESCRIPTION
**Pull request by antrix190**
# Description

This PR adds star/unstar a repository for an authenticated user.

Fixes : #1354 

# Before submitting a PR:

- [x] Changes must not break binary backwards compatibility. If you are unclear on how to make the change you think is needed while maintaining backward compatibility, [CONTRIBUTING.md](CONTRIBUTING.md) for details.
- [x] Add JavaDocs and other comments as appropriate. Consider including links in comments to relevant documentation on https://docs.github.com/en/rest .  
- [x] Add tests that cover any added or changed code. This generally requires capturing snapshot test data. See [CONTRIBUTING.md](CONTRIBUTING.md) for details.
- [x] Run `mvn -D enable-ci clean install site` locally. If this command doesn't succeed, your change will not pass CI.
- [x] Push your changes to a branch other than `main`. You will create your PR from that branch.

# When creating a PR: 

- [x] Fill in the "Description" above with clear summary of the changes. This includes:
  - [x] If this PR fixes one or more issues, include "Fixes #<issue number>" lines for each issue. 
  - [x] Provide links to relevant documentation on https://docs.github.com/en/rest where possible.
- [x] All lines of new code should be covered by tests as reported by code coverage. Any lines that are not covered must have PR comments explaining why they cannot be covered. For example, "Reaching this particular exception is hard and is not a particular common scenario."
- [x] Enable "Allow edits from maintainers".

### CODE REVIEW

File: src/main/java/org/kohsuke/github/GHRepository.java [hunk 1]
```diff
@@ -2981,6 +2981,13 @@ String getApiTailUrl(String tail) {
         return "/repos/" + getOwnerName() + "/" + name + tail;
     }
 
+    String getApiHeadUrl(String head) {
+        if (head.length() > 0 && !head.endsWith("/")) {
+            head = '/' + head;
+        }
+        return "/user/" + head + getOwnerName() + "/" + name;
+    }
+
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHRepository.java:2990]: I don't think this is needed here.  Each of the methods below can hard code `"/user/starred/" + full_name`.

**Review by bitwiseman — CHANGES_REQUESTED** [review]
Good start.  Just need tests.

File: src/main/java/org/kohsuke/github/GHRepository.java (hunk 1 above)
Comment (antrix190) [reply src/main/java/org/kohsuke/github/GHRepository.java:3053]: Hey @bitwiseman, I will work on this issue now. Bit lost in creating the test data for test cases, but i will soon figure it out.

**Comment by antrix190** [comment]
@bitwiseman can you trigger build for this PR.

File: src/main/java/org/kohsuke/github/GHRepository.java (hunk 1 above)
Comment (antrix190) [reply src/main/java/org/kohsuke/github/GHRepository.java]: Fixed.

**Comment by antrix190** [comment]
@bitwiseman Can you have a look at the PR, have addressed all the open comments

**Review by bitwiseman — APPROVED** [review]

**Pull request merged**